src/app/main/component/user/components/profile/profile-cards/profile-cards.component.ts

```diff
@@ -23,8 +26,27 @@ export class ProfileCardsComponent implements OnInit, OnDestroy {
 
   ngOnInit() {
     this.languageSubscription = this.localStorageService.languageBehaviourSubject.subscribe(() => {
+      this.loadFactsFromLocalStorage();
       this.getFactOfTheDay();
+      this.checkAndUpdateHabitFact();
     });
+
+    this.dailyUpdateSubscription = timer(0, 24 * 60 * 60 * 1000).subscribe(() => {
+      this.checkAndUpdateHabitFact();
+    });
+  }
+
+  loadFactsFromLocalStorage() {
+    const savedHabitFact = localStorage.getItem('habitFactOfTheDay');
+    const lastHabitFetchTime = localStorage.getItem('lastHabitFactFetchTime');
+    const currentTime = Date.now();
+    const oneDay = 24 * 60 * 60 * 1000;
```

Comment:
repeated more then one time use of this expression, would it be better to create a global const from it?